e2e/tests/plugins/timer/timer.e2e.spec.js

```diff
@@ -0,0 +1,156 @@
+        await page.click('text=Timer');
+
+        // Click text=OK
+        await Promise.all([
+            page.waitForNavigation({waitUntil: 'networkidle'}),
+            page.click('text=OK')
+        ]);
+
+        await expect(page.locator('.l-browse-bar__object-name')).toContainText('Unnamed Timer');
+    });
+
+    test('Can start, pause, restart, and stop timer within context menu and 3dot menu', async ({ page }) => {
+        test.info().annotations.push({
+            type: 'issue',
+            description: 'https://github.com/nasa/openmct/issues/4313'
+        });
+
+        await test.step("Use Timer actions from the tree context menu", async () => {
+            await triggerTimerContextMenuAction(page, 'Start');
+            await triggerTimerContextMenuAction(page, 'Pause');
+            await triggerTimerContextMenuAction(page, 'Restart at 0');
+            await triggerTimerContextMenuAction(page, 'Stop');
+        });
+
+        await test.step("Use Timer actions from the 3dot menu", async () => {
+            await triggerTimer3dotMenuAction(page, 'Start');
+            await triggerTimer3dotMenuAction(page, 'Pause');
+            await triggerTimer3dotMenuAction(page, 'Restart at 0');
+            await triggerTimer3dotMenuAction(page, 'Stop');
+        });
+    });
+});
+
+/**
+ * Open the timer context menu from the object tree.
+ * Expands the 'My Items' folder if it is not already expanded.
+ * @param {import('@playwright/test').Page} page
+ */
+async function openTimerContextMenu(page) {
+    const myItemsFolder = page.locator('text=Open MCT My Items >> span').nth(3);
+    const className = await myItemsFolder.getAttribute('class');
+    if (!className.includes('c-disclosure-triangle--expanded')) {
+        await myItemsFolder.click();
+    }
+
+    await page.locator(`a:has-text("Unnamed Timer")`).click({
+        button: 'right'
+    });
+}
+
+/**
+ * Trigger a timer action from the tree context menu
+ * @param {import('@playwright/test').Page} page
+ * @param {'Start' | 'Stop' | 'Pause' | 'Restart at 0'} action
+ */
+async function triggerTimerContextMenuAction(page, action) {
+    await openTimerContextMenu(page);
+    let menuOptions = page.locator('.c-menu ul');
+    await expect.soft(menuOptions).toContainText(action);
+    await page.locator(`.c-menu ul li:has-text("${action}")`).click();
+    await openTimerContextMenu(page);
+    menuOptions = page.locator('.c-menu ul');
+    await assertAvailableActions(menuOptions, action);
+}
+
+/**
+ * Trigger a timer action from the 3dot menu
+ * @param {import('@playwright/test').Page} page
+ * @param {'Start' | 'Stop' | 'Pause' | 'Restart at 0'} action
+ */
+async function triggerTimer3dotMenuAction(page, action) {
+    const threeDotMenuButton = 'button[title="More options"]';
+    await page.click(threeDotMenuButton);
+    await page.locator(`.c-menu ul li:has-text("${action}")`).click();
+
+    // FIXME: Figure out a way to not need an explicit wait here.
```

Comment:
Hmm. Let's leave this for now and just be mindful of the stopgap.

I'm weary of anything tied to listing off elements of text in the menu to compare.

Reply:
Yeah, I wonder if this test is trying to do too much. It should be enough that it just tries to execute the action. I'm going to revisit this and see if I can simplify.

Still need to find a way to avoid the hard wait, though.